src/content/app/regulatory-activity-viewer/state/api/activityViewerApiSlice.ts

```diff
@@ -16,53 +16,109 @@
 
 import restApiSlice from 'src/shared/state/api-slices/restSlice';
 
+import config from 'config';
+
 import type { OverviewRegion } from 'src/content/app/regulatory-activity-viewer/types/regionOverview';
 import type { Epigenome } from 'src/content/app/regulatory-activity-viewer/types/epigenome';
 import type { EpigenomeMetadataDimensionsResponse } from 'src/content/app/regulatory-activity-viewer/types/epigenomeMetadataDimensions';
+import type { EpigenomeActivityResponse } from 'src/content/app/regulatory-activity-viewer/types/epigenomeActivity';
+
+export type Location = {
+  regionName: string;
+  start: number;
+  end: number;
+};
+
+type RegionOverviewRequestParams = {
+  assemblyName: string; // <-- this will be replaced by assembly accession id
+  location: string; // <-- as formatted by the stringifyLocation function
+};
+
+type BaseEpigenomesRequestParams = {
+  assemblyName: string; // <-- this will be replaced by assembly accession id
+};
+
+type EpigenomeMetadataRequestParams = {
+  assemblyName: string; // <-- this will be replaced by assembly accession id
+};
+
+type EpigenomesActivityRequestParams = {
+  assemblyAccessionId: string;
+  regionName: string;
+  locations: { start: number; end: number }[];
+  epigenomeIds: string[];
+};
+
+export const stringifyLocation = (location: Location) =>
+  `${location.regionName}:${location.start}-${location.end}`;
 
 const activityViewerApiSlice = restApiSlice.injectEndpoints({
   endpoints: (builder) => ({
-    regionOverview: builder.query<OverviewRegion, void>({
-      queryFn: async () => {
-        // const module = await import(
-        //   'tests/fixtures/activity-viewer/mockRegionOverviewSparse'
-        // );
-        const module = await import(
-          'tests/fixtures/activity-viewer/mockRegionOverviewDense'
-        );
-        const data = module.default;
-
-        return { data };
+    regionOverview: builder.query<OverviewRegion, RegionOverviewRequestParams>({
+      query: (params) => {
+        const { assemblyName, location } = params;
+        return {
+          url: `${config.regulationApiBaseUrl}/region-of-interest/v0.1.0/${assemblyName}?location=${location}`
```

Comment:
Shouldn't the version (`v0.1.01) be a constant or an environment variable?

Reply:
Mmm, it isn't our constant/variable; it's regulation's. And we don't reuse it anywhere other than in this url string. I think it's too early to worry about those; we will see how regulation develop their apis and how they will agree with Platforms to expose their endpoints.